lib/fluent/config/literal_parser.rb

```diff
@@ -254,8 +254,15 @@ def scan_json(is_array)
               buffer << line_buffer + "\n"
               line_buffer = ""
             else
-              # '#' is a char in json string
-              line_buffer << char
+              if @ss.exist?(/\{[^}]+\}/)
```

Comment:
```suggestion
              if @ss.exist?(/^\{[^}]+\}/)
```
Need to  match `{` just after `#`.